packages/shared/src/components/cards/SourceButton.tsx

```diff
@@ -1,33 +1,35 @@
 import React, { CSSProperties, ReactElement } from 'react';
 import classNames from 'classnames';
-import { Post } from '../../graphql/posts';
 import { TooltipPosition } from '../tooltips/BaseTooltipContainer';
 import { LinkWithTooltip } from '../tooltips/LinkWithTooltip';
+import { Source } from '../../graphql/sources';
+
+interface SourceButtonProps {
+  source: Source;
+  className?: string;
+  style?: CSSProperties;
+  tooltipPosition?: TooltipPosition;
+}
 
 export default function SourceButton({
-  post,
+  source,
   className,
   style,
   tooltipPosition = 'bottom',
-}: {
-  post: Post;
-  className?: string;
-  style?: CSSProperties;
-  tooltipPosition?: TooltipPosition;
-}): ReactElement {
-  return (
+}: SourceButtonProps): ReactElement {
+  return source ? (
     <LinkWithTooltip
-      href={`${process.env.NEXT_PUBLIC_WEBAPP_URL}sources/${post.source.id}`}
+      href={`${process.env.NEXT_PUBLIC_WEBAPP_URL}sources/${source.id}`}
       prefetch={false}
-      tooltip={{ content: post.source.name, placement: tooltipPosition }}
+      tooltip={{ content: source.name, placement: tooltipPosition }}
     >
       <a className={classNames('flex cursor-pointer', className)} style={style}>
         <img
-          src={post.source.image}
-          alt={post.source.name}
-          className="w-6 h-6 rounded-full bg-theme-bg-tertiary"
+          src={source.image}
+          alt={source.name}
+          className="w-8 h-8 rounded-full bg-theme-bg-tertiary"
         />
       </a>
```

Comment:
I think we can actually use the ProfilePicture component for this, we use it in other places for the a rounded source as well.

Reply:
Totally agree. Let me work this out 🚢